### DISCUSSION TITLE
Reason for only having AddMonths (but no other components) in GameClockDateTime?

### DISCUSSION THREAD

**Discussion by nomakewan** [scripthookvdotnet/scripthookvdotnet#1346] [Q&A] [answered]
Not sure whether to file this under Bug Report or Feature Request, so will consider it a Feature Request.

I was curious as to why GameClockDateTime has an AddMonths function but not any of its other components (AddYears, AddDays, AddHours, AddMinutes, AddSeconds). Is there a particular design reason for this? If not, then would it be possible to add these functions in?

Thanks as always!

EDIT: Thanks for moving; I forgot there's a Discussions section now. Not used to that on Github yet.

**Comment by kagikn** [comment, marked as answer]
`GameClockDate` and `GameClockDateTime` have `AddMonths` because you can't really calculate the amount of seconds by just specifying that of one month without any ambiguity or any additional constraints, and `GameClockDuration` is not responsible to be aware of how amount of month varies by month number. While you can calculate fixed amount of seconds by just specifying amount of minutes, hours, days, and weeks, and you can add or subtract a date time by a duration without the ambiguity. `AddMonths` guarantees that it uses the last day of the month if the day does not exist in the resulting month just like how [`NaiveDate::checked_add_months`](https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDate.html#method.checked_add_months) in the `chrono` in Rust guarantees (so it has an additional constraint).

So, when you add or subtract time from `GameClockDateTime`:
* In General: add or subtract a `GameClockDutation`
* For month diff: use `AddMonths`
* For year diff: use `GameClockDate.FromYmd` with a different year (be careful with Feb 29)

It would be appreciated if your questions where nothing is clearly actionable are posted in the Discussions page (like questions about the idea behind how specific classes are designed). I'm using the feature like this. I'd convert discussions that are clearly actionable into issues if I find ones.
